### ISSUE TITLE
[Bug] IsAppServicesAadAuthenticationEnabled not working with new App Service Authentication flow

### ISSUE THREAD

**Issue by B742** [AzureAD/microsoft-identity-web#1163]
**Which version of Microsoft Identity Web are you using?**
   Microsoft Identity Web 1.9.1 with ASP.NET 5.0

**Where is the issue?**
* Web app
    * [ ] Sign-in users
    * [ ] Sign-in users and call web APIs
* Web API
    * [ ] Protected web APIs (validating tokens)
    * [ ] Protected web APIs (validating scopes)
    * [ ] Protected web APIs call downstream web APIs
* Token cache serialization
     * [ ] In-memory caches
     * [ ] Session caches
     * [ ] Distributed caches
* Other (please describe)
   Identity provider detection

**Is this a new or an existing app?**
   Existing App using Azure App Service and Easy auth. Switched from Authentication (classic) to the new Authentication experience.

**Repro**

Use the following in the Startup.cs:
```csharp
public void ConfigureServices(IServiceCollection services)
{
    ...
    services.AddMicrosoftIdentityWebAppAuthentication(Configuration);
    ...
}
public void Configure(IApplicationBuilder app)
{
    ...
    app.UseAuthentication();
    app.UseAuthorization();
    ...
}
```
Push it to an App Service with the new Authentication experience enabled using AAD. Then try to call the API, you will get an exception saying that you need to provide the ClientId and etc. These values were not needed before the switch to the new experience. 

**Expected behavior**
The library should detect that App Service's Authentication is enabled and not ask for "AzureAd" configuration section.

**Actual behavior**
The ASP.NET 5.0 server throws:
`[Error] Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware: An exception was thrown attempting to execute the error handler.Microsoft.Extensions.Options.OptionsValidationException: IDW10106: The 'ClientId' option must be provided.at`

It should not.

**Possible solution**
In AppServicesAuthenticationInformation.IsAppServicesAadAuthenticationEnabled, there is 2 conditions to be met.
The first one is still valid when using the new Authentication experience. However, the second one, the string needs to match the variable"AppServicesAuthAzureActiveDirectory" value which is "AzureActiveDirectory". However, using the new experience, the environment variable value is "azureactivedirectory". Since the equal is case-sensitive, it does not work.

**Additional context / logs / screenshots**
When using classic experience:
![oldexperience](https://user-images.githubusercontent.com/23298800/116306743-76bb5c00-a773-11eb-80b1-6ecec5e311ac.PNG)

When using the new experience:
![newexperience](https://user-images.githubusercontent.com/23298800/116306761-7d49d380-a773-11eb-9632-2dd6b3b99fe8.PNG)

**Comment by jmprieur** [comment]
@jennyf19 : we need to do insensitive comparisons of the environment variables ... apparently Easy auth changed the values;

Thanks @navyasric for looking at this issue.

**Comment by falconkirtaran** [comment]
For what it's worth, in my completely randomly drawn Azure App Service, after migrating from classic authentication, the value of the WEBSITE_AUTH_DEFAULT_PROVIDER environment variable was neither "AzureActiveDirectory" nor "azureactivedirectory", but rather, "aad".

Pending the update, I was able to crowbar it into correct functioning by adding this to the top of ConfigureServices, which I'm greatly looking forward to removing:

            var AuthProviderEnv = Environment.GetEnvironmentVariable("WEBSITE_AUTH_DEFAULT_PROVIDER");
            if (string.Equals(AuthProviderEnv, "AzureActiveDirectory", StringComparison.InvariantCultureIgnoreCase) ||
                string.Equals(AuthProviderEnv, "aad", StringComparison.InvariantCultureIgnoreCase))
            {
                Environment.SetEnvironmentVariable("WEBSITE_AUTH_DEFAULT_PROVIDER", "AzureActiveDirectory");
            }

**Comment by jmprieur** [comment]
@falconkirtaran : this is even worse than I thought.
So how to know which value is used for the environment variable ? I'm not a big fan of having a setting, for something that you'd expect would just work.

cc: @jennyf19 @navyasric @bgavrilMS

**Comment by jennyf19** [comment]
original issue Included in [1.9.2 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.9.2) -> do we make a new issue? not really an id web issue, right?

**Comment by jmprieur** [comment]
We need to have a look at the EasyAuth experience.
I'd think we'd want to look for "aad" as well :(

**Comment by jmprieur** [comment]
Reopening as EasyAuth people told us that EasyAuth accepts either azureactivedirectory or aad case insensitive when specifying the default provider. This is set by whatever client (likely the portal) is configuring the customer's apps.

**Comment by jennyf19** [comment]
Fixed in [1.13 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.13.0).
